prow/config/jobs.go

```diff
@@ -334,12 +345,38 @@ func (ps Postsubmit) ShouldRun(baseRef string, changes ChangedFilesProvider) (bo
 	if !ps.CouldRun(baseRef) {
 		return false, nil
 	}
+	// If the `always_run` field was explicitly set to true, we should run it.
+	// There's no need to consider other possibilities like `run_if_changed` or
+	// `skip_if_only_changed` (and even if we tried to set either of these
+	// fields with `always_run: true`, it would have been an invalid
+	// configuration anyway).
+	if ps.AlwaysRun != nil && *ps.AlwaysRun {
+		return true, nil
+	}
+
+	// At this point either `always_run: false` is there, or one of
+	// `run_if_changed` or `skip_if_only_changed` could be set. The definition
+	// of `run_if_changed` or `skip_if_only_changed` rules take precedence, so
+	// consider them first.
 	if determined, shouldRun, err := ps.RegexpChangeMatcher.ShouldRun(changes); err != nil {
 		return false, err
 	} else if determined {
 		return shouldRun, nil
 	}
-	// Postsubmits default to always run
+
+	// At this point neither `run_if_changed` nor `skip_if_only_changed` were
+	// set. We're left with 2 cases: (1) `always_run: false` was provided, or
+	// (2) this field was not defined in the job at all. Only in the second case
+	// do we return "true".
+
+	// If the `always_run` field was explicitly set to false, override the
+	// default `true` response.
+	if ps.AlwaysRun != nil && !*ps.AlwaysRun {
+		return false, nil
+	}
```

Comment:
nit: If we change this to 
```suggestion
	if ps.AlwaysRun != nil {
		return *ps.AlwaysRun, nil
	}
```
we don't need the other AlwaysRun check before the regexp change check. I think that will make this easier to understand.

Reply:
Agreed. FTR the only reason I did it the current way is because I wanted to align as closely as possible to the behavior of `ShouldRun` for Presubmits. There we perform an `if ps.AlwaysRun` check *before* we examine `ps.RegexpChangeMatcher` (see line 392). I assume your suggestion still stands?